core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1617,7 +1615,7 @@ public void receivePending(Transaction tx, @Nullable List<Transaction> dependenc
             // If this tx spends any of our unspent outputs, mark them as spent now, then add to the pending pool. This
             // ensures that if some other client that has our keys broadcasts a spend we stay in sync. Also updates the
             // timestamp on the transaction and registers/runs event listeners.
-            commitTx(tx);
+            commitTx(tx, false);
```

Comment:
Each mempool is different. You'll get connected to different peers. The transaction can have arrived through other channels, e.g. Bluetooth. It's important the transaction gets saved to the wallet, such that it can be rebroadcasted to connecting peers.